core/src/main/scala/org/apache/spark/rdd/PipedRDD.scala

```diff
@@ -144,7 +142,8 @@ private[spark] class PipedRDD[T: ClassTag](
     new Thread(s"stdin writer for $command") {
       override def run(): Unit = {
         TaskContext.setTaskContext(context)
-        val out = new PrintWriter(proc.getOutputStream)
+        val out = new PrintWriter(new BufferedWriter(
```

Comment:
@srowen  - Thanks for taking a look. In our testing we found out that using a buffer of large size (1 MB) gives us a cpu savings of around 15%. It makes sense to be able to increase the buffer size when we are piping a large amount of data. If changing a public API is not too much trouble, it would be pretty useful for us to have a configurable buffer size. 

Regarding your second point, I am not sure if I understand you. My change is not going to change the behavior of the PrintWriter at all.  Do you mean to say the issue with UTF-8 encoding already exists and I should fix it in this diff?